source/fundamentals/serialization/poco.txt

```diff
@@ -203,26 +250,303 @@ The following code sample maps the ``Identifier`` property to the
+   });
+
+.. tip:: Specify an ``IIdGenerator`` for Multiple Classes
+
+   You can use the ``RegisterIdGenerator()`` method to specify a single ``IIdGenerator``
+   for all ``Id`` properties of a certain data type. The following code example instructs
+   the driver to use the ``CombGuidGenerator`` type for all ``Guid`` IDs:
+
+   .. code-block:: csharp
+      :copyable: true
+   
+      BsonSerializer.RegisterIdGenerator(
+        typeof(Guid),
+        CombGuidGenerator.Instance
+      );
+
+The {+driver-short+} also includes ``IIdGenerator`` types that validate the ``Id``
+property and throw an exception if the ID is invalid. The following table lists these
+types:
+
+.. list-table::
+   :header-rows: 1
+   :widths: 10 10 
+
+   * - ID Validation 
+     - ``IIdGenerator`` Type 
+   * - Not null 
+     - ``NullIdChecker``
+   * - Not all zeroes 
+     - ``ZeroIdChecker<T>``
+
+In the following code example, if the ``Id`` property of the ``House`` class contains
+the default value (``null``), the driver will throw an exception:
+
+.. code-block:: csharp
+   :copyable: false
+   :emphasize-lines: 3
+   
+   public class House
+   {
+       [BsonId(IdGenerator = typeof(NullIdChecker))]
+       public Guid Id { get; set; }
+   }
+
+Customize DateTime Serialization
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To customize how the {+driver-short+} serializes ``DateTime`` properties, use the
+``[BsonDateTimeOptions()]`` attribute and specify the desired setting as an argument.
+
+If a ``DateTime`` property represents only a date, you can apply the
+``[BsonDateTimeOptions(DateOnly = true)]`` attribute to it.
+If you do, the driver won't perform time-zone conversion on the value.
+
+In the following code example, the ``PatientRecord`` class uses a ``DateTime`` for the
+``DateOfBirth`` property. The ``[BsonDateTimeOptions(DateOnly = true)]``
+attribute indicates that the property contains only a date.
+
+.. code-block:: csharp
+   :copyable: true
+   :emphasize-lines: 5
+
+   public class PatientRecord
+   {
+       public Guid Id { get; set; }
+
+       [BsonDateTimeOptions(DateOnly = true)]
+       public DateTime DateOfBirth { get; set; }
+   }
+
+You can also use the ``[BsonDateTimeOptions()]`` attribute to specify the ``DateTimeKind``
+of a ``DateTime`` property. In the following code example, the ``PatientRecord`` class
+uses a ``DateTime`` for the ``AppointmentTime`` property.
+The ``[BsonDateTimeOptions(Kind = DateTimeKind.Local)]`` attribute indicates
+that the time component of the property's value is in local time. When the
+driver serializes this property, it will convert the time to UTC, the standard
+format for times stored in MongoDB.
+
+.. code-block:: csharp
+   :copyable: true
+   :emphasize-lines: 8
+
+   public class PatientRecord
+   {
+       public Guid Id { get; set; }
+
+       [BsonDateTimeOptions(DateOnly = true)]
+       public DateTime DateOfBirth { get; set; }
```

Comment:
Q: Does this example need this field if its demonstrating the `Kind` setting?

Reply:
No, I just kept it as a continuation of the previous one. Is it clearer if I leave it out?

Reply:
yes, I think it could be distracting. So far, this page doesn't use the same exact class in each section, should not be confusing to the reader if this class only demonstrates the attribute of note.